windows/src/test/test-kmcomapi-win64/Keyman.Test.Console.KeymanAPIHost.pas

```diff
@@ -0,0 +1,435 @@
+﻿unit Keyman.Test.Console.KeymanAPIHost;
+
+interface
+
+uses
+  DUnitX.TestFramework,
+  Winapi.Windows,
+  Winapi.ActiveX,
+  System.Win.ComObj,
+  System.SysUtils,
+  keymanapi_tlb;
+
+type
+  [TestFixture]
+  TKeymanAPITest = class
+  private
+    FTestPath: string;
+    k: IKeyman;
+  public
+    constructor Create;
+    destructor Destroy; override;
+
+    [Test]
+    procedure Test_IKeyman;
+    [Test]
+    procedure Test_IKeyman_Apply;
+    [Test]
+    procedure Test_IKeyman_Refresh;
+
+    [Test]
+    procedure Test_IKeymanCollection;
+
+    [Test]
+    procedure Test_IKeymanErrors;
+
+    [Test]
+    procedure Test_IKeymanHotkeys;
+
+    [Test]
+    procedure Test_IKeymanKeyboardLanguagesInstalled;
+
+    [Test]
+    procedure Test_IKeymanKeyboardOptions;
+
+    [Test]
+    procedure Test_IKeymanKeyboards;
+
+    [Test]
+    procedure Test_IKeymanKeyboardsInstalled;
+
+    [Test]
+    procedure Test_IKeymanPackageContentKeyboards;
+
+    [Test]
+    procedure Test_IKeymanLanguages;
+
+    [Test]
+    procedure Test_IKeymanOptions;
+
+    [Test]
+    procedure Test_IKeymanPackageContentFiles;
+
+    [Test]
+    procedure Test_IKeymanPackageContentFonts;
+
+    [Test]
+    procedure Test_IKeymanPackagesInstalled;
+
+    [Test]
+    procedure Test_IKeymanControl;
+
+    [Test]
+    procedure Test_IKeymanError;
+
+    [Test]
+    procedure Test_IKeymanHotkey;
+
+    [Test]
+    procedure Test_IKeymanKeyboard;
+
+    [Test]
+    procedure Test_IKeymanKeyboardFile;
+
+    [Test]
+    procedure Test_IKeymanKeyboardInstalled;
+
+    [Test]
+    procedure Test_IKeymanKeyboardLanguageInstalled;
+
+    [Test]
+    procedure Test_IKeymanKeyboardOption;
+
+    [Test]
+    procedure Test_IKeymanLanguage;
+
+    [Test]
+    procedure Test_IKeymanOption;
+
+    [Test]
+    procedure Test_IKeymanPackage;
+
+    [Test]
+    procedure Test_IKeymanPackageFile;
+
+    [Test]
+    procedure Test_IKeymanPackageInstalled;
+
+    [Test]
+    procedure Test_IKeymanPackageContentFile;
+
+    [Test]
+    procedure Test_IKeymanPackageContentFont;
+
+    [Test]
+    procedure Test_IKeymanSystemInfo;
+
+    [Test]
+    procedure Test_IKeymanUserInterface;
+
+    [Test]
+    procedure Test_IKeymanVisualKeyboard;
+  end;
+
+implementation
+
+constructor TKeymanAPITest.Create;
+begin
+  inherited Create;
+  FTestPath := ExtractFilePath(ExtractFileDir(ExtractFileDir(ParamStr(0)))) + 'test\';
+  CoInitializeEx(nil, COINIT_APARTMENTTHREADED);
+  k := CoKeyman.Create;
+end;
+
+destructor TKeymanAPITest.Destroy;
+begin
+  k := nil;
+  CoUninitialize;
+  inherited Destroy;
+end;
+
+//
+// IKeyman
+//
+
+procedure TKeymanAPITest.Test_IKeyman;
+begin
+  // Implicitly tests each interface _Add/_Release as well,
+  // verifying that they are legit objects
+  Assert.IsNotNull(k.Keyboards);
```

Comment:
Based on my reading of the code here, `k` is initialized once, before all the tests of `TKeymanAPITest` are run, rather than before each test is run to ensure a fresh instance of `k`?

Reply:
True. It might be better to put `k := CoKeyman.Create;` into a `Setup` method.